### ISSUE TITLE
Connection closed against the specified behaviour of HTTP 1.1

### ISSUE THREAD

**Issue by ChrisGreenaway** [lipanski/mockito#103]
In the HTTP 1.1 specification, the default is to keep the connection open after processing a request.

The spec says:

> If the server chooses to close the connection immediately after sending the response, it SHOULD send a Connection header including the connection-token close.

However mockito closes the connection after processing every request.

It causes the following test to fail (dependent on machine / timing):

```
#[test]
fn http_fail() {
    let _mock_server = mock("GET", "/s").with_status(200).with_body("b").create();

    let mut executor = create_exec();
    let client = hyper::Client::new();
    let url = format!("{}/s", &mockito::server_url());

    let mut pass = true;

    for i in 0..10000 {
        let uri = url.parse::<hyper::Uri>().unwrap();
        let request = client.get(uri);
        if let Err(response_error) = executor.block_on(request) {
            println!("Failed {}: {} : {:?}", i, response_error, response_error);
            pass = false
        }
    }

    assert!(pass);
}

fn create_exec() -> tokio::runtime::Runtime {
    tokio::runtime::Builder::new()
        .basic_scheduler()
        .enable_time()
        .enable_io()
        .build()
        .unwrap()
}
```

**Comment by ChrisGreenaway** [comment]
Possible workaround: use `.with_header("Connection", "close")` on every mock.

**Comment by lipanski** [comment]
@ChrisGreenaway something like this https://github.com/lipanski/mockito/pull/104 ?

**Comment by ChrisGreenaway** [comment]
@lipanski - that's a change that should make my test pass.

Of course, the question this issue suggests is: what happens if someone wants to test with `Connection: keep-alive`. They can set the header on the mock, but the connection will get closed anyway. I appreciate that fixing that would require quite a substantial change to how connections are handled and moves this library substantially towards having to fully understand the nuances of all the HTTP protocol options / versions, so it's not something I'm expecting from this issue, but I suspect you'll get future issues that need that kind of change.

**Comment by lipanski** [comment]
yeah, I was thinking about that but indeed it would require changing the HTTP client implementation more significantly. when the `keep-alive` discussion arises, we can revist. for now I'll just deploy the default header fix. thanks for reporting the issue :1st_place_medal:

**Comment by lipanski** [comment]
released as 0.23.1
